src/hooks/useActionAddCol.js

```diff
@@ -0,0 +1,20 @@
+import { GridCell } from "../util/generate_grid_nodes";
+
+const useActionAddCol = (gridNodes, setGridNodes) => {
+  const handleSelect = () => {
+    if (!gridNodes.length) {
+        gridNodes.push([]);
```

Comment:
Not generally a good practice to modify state directly. You should instead create another `gridNodes` array variable which would be a clone of the original `gridNodes`.

You probably won't face any issue related to that in this application cause it's pretty small but the main reason for that would be race condition. If another function was also working with `gridNodes` at the time that this was being executed, there would be corrupted data.

Reply:
That makes a lot of sense. Thanks for the explanation!